### PR TITLE
systemctl: port to extract_first_word

### PR DESCRIPTION
**Pull request by ssahani**

### CODE REVIEW

File: src/systemctl/systemctl.c [hunk 1]
```diff
@@ -6693,18 +6695,21 @@ static int systemctl_parse_argv(int argc, char *argv[]) {
...
+                                                return log_error_errno(r, "Failed to parse property: %s", optarg);
 
-                                        prop = strndup(word, size);
-                                        if (!prop)
-                                                return log_oom();
+                                        if (r == 0)
+                                                break;
 
-                                        if (strv_consume(&arg_properties, prop) < 0)
+                                        if (strv_push(&arg_properties, prop) < 0)
```
Comment (filbranden) [inline src/systemctl/systemctl.c:6712]: Why not just this?

```
  if (isempty(optarg) && !arg_properties) {
    ...
  } else {
    arg_properties = strv_split(optarg, ",");
  }
```

It turns out `strv_split` still uses `FOREACH_WORD_SEPARATOR` but we can fix that to use `extract_first_word` too...

Cheers!
Filipe

Comment (ssahani) [inline src/systemctl/systemctl.c:6712]: else is not required

```
  if (isempty(optarg) && !arg_properties)
                                return log_error_errno(r, "--type requires arguments.");
```

yes strv_split uses twice .
1. to find how many to allocate
2. then actually it stores into them

which will look same as strv_split_extract if we replace with extract_first_word
except ExtractFlags

**Comment by zonque** [comment]
Don't you see this `Assertion 'r == r_expected' failed at src/test/test-fstab-util.c:48, function do_fstab_filter_options(). Aborting.` error locally when running `make check`? Because I do, and so does Semaphore.

**Comment by ssahani** [comment]
I am not too sure why the build is failing here. what is has to do with do_fstab_filter_options

**Comment by zonque** [comment]
> I am not too sure why the build is failing here. what is has to do with do_fstab_filter_options

Again: does `make check` work for you, on your local machine?

**Pull request closed without merging**